src/topojson/topojson.ml

```diff
@@ -404,12 +404,18 @@ module Make (J : Intf.Json) = struct
       | Error _ -> []
 
     let base_of_json json =
-      match
-        ( J.find json [ "objects" ],
-          J.find json [ "arcs" ],
-          J.find json [ "transform" ] )
-      with
-      | Some objects, Some arcs, Some transform ->
+      match (J.find json [ "scale" ], J.find json [ "translate" ]) with
+      | Some scale, Some translate ->
+          let* scale = J.to_array (decode_or_err J.to_float) scale in
+
+          let* translate = (J.to_array (decode_or_err J.to_float)) translate in
+
+          Ok { scale; translate }
+      | _, _ -> Error (`Msg "No transform field in Topology object!")
```

Comment:
This is not an error, the transform object is optional.

Reply:
Okay , so what should 
```ocaml
_ , _  ->
``` 
return?

Reply:
The error is ok actually but it being applied incorrectly. It is trying to read the fields of the transform object so if scale and translate are not present it is indeed an error (but the message is wrong). Where this function (confusingly named `base_of_json`) is used is not reading the correct JSON which is why it isn't right.

Reply:
Okay. So I need to rename it to `tranform_of_json` as you said in the earlier comment below. But I'm still not clear on what the function is to do.

I saw [here](https://github.com/topojson/topojson-client#transform)
> [#](https://github.com/topojson/topojson-client#transform) topojson.transform(transform) [<>](https://github.com/topojson/topojson-client/blob/master/src/transform.js)
If the specified [transform object](https://github.com/topojson/topojson-specification/blob/master/README.md#212-transforms) is non-null, returns a [point transform function](https://github.com/topojson/topojson-client#_transform) to remove delta-encoding and apply the transform. If the transform is null, returns the identity function.

And [point transform function](https://github.com/topojson/topojson-client#_transform) says 
>[#](https://github.com/topojson/topojson-client#_transform) transform(point[, index])
Applies this transform function to the specified point, returning a new point with the transformed coordinates. If the specified index is truthy, the input point is treated as relative to the previous point passed to this transform, as is the case with delta-encoded arcs

Is this in line with what `tranform_of_json` is supposed to do?

Reply:
Okay